$OpenBSD: patch-src_backends_native_meta-cursor-renderer-native_c,v 1.1 2014/10/15 13:35:27 jasper Exp $

Revert 488dd0b402dc8c56a70f995a7456c2ff511a818b,
while Wayland-specific, it requires Mesa >= 10.3.

--- src/backends/native/meta-cursor-renderer-native.c.orig	Thu Oct  2 16:29:20 2014
+++ src/backends/native/meta-cursor-renderer-native.c	Wed Oct 15 13:59:39 2014
@@ -27,27 +27,16 @@
 #include "meta-cursor-renderer-native.h"
 
 #include <gbm.h>
-#include <xf86drm.h>
 
 #include "meta-cursor-private.h"
 #include "meta-monitor-manager.h"
 
-#ifndef DRM_CAP_CURSOR_WIDTH
-#define DRM_CAP_CURSOR_WIDTH 0x8
-#endif
-#ifndef DRM_CAP_CURSOR_HEIGHT
-#define DRM_CAP_CURSOR_HEIGHT 0x9
-#endif
-
 struct _MetaCursorRendererNativePrivate
 {
   gboolean has_hw_cursor;
 
   int drm_fd;
   struct gbm_device *gbm;
-
-  uint64_t cursor_width;
-  uint64_t cursor_height;
 };
 typedef struct _MetaCursorRendererNativePrivate MetaCursorRendererNativePrivate;
 
@@ -82,13 +71,17 @@ set_crtc_cursor (MetaCursorRendererNative *native,
     {
       struct gbm_bo *bo;
       union gbm_bo_handle handle;
+      int width, height;
       int hot_x, hot_y;
 
       bo = meta_cursor_reference_get_gbm_bo (cursor, &hot_x, &hot_y);
 
       handle = gbm_bo_get_handle (bo);
+      width = gbm_bo_get_width (bo);
+      height = gbm_bo_get_height (bo);
+
       drmModeSetCursor2 (priv->drm_fd, crtc->crtc_id, handle.u32,
-                         priv->cursor_width, priv->cursor_height, hot_x, hot_y);
+                         width, height, hot_x, hot_y);
     }
   else
     {
@@ -193,19 +186,6 @@ meta_cursor_renderer_native_init (MetaCursorRendererNa
       CoglRenderer *cogl_renderer = cogl_display_get_renderer (cogl_context_get_display (ctx));
       priv->drm_fd = cogl_kms_renderer_get_kms_fd (cogl_renderer);
       priv->gbm = gbm_create_device (priv->drm_fd);
-
-      uint64_t width, height;
-      if (drmGetCap (priv->drm_fd, DRM_CAP_CURSOR_WIDTH, &width) == 0 &&
-          drmGetCap (priv->drm_fd, DRM_CAP_CURSOR_HEIGHT, &height) == 0)
-        {
-          priv->cursor_width = width;
-          priv->cursor_height = height;
-        }
-      else
-        {
-          priv->cursor_width = 64;
-          priv->cursor_height = 64;
-        }
     }
 #endif
 }
@@ -216,16 +196,6 @@ meta_cursor_renderer_native_get_gbm_device (MetaCursor
   MetaCursorRendererNativePrivate *priv = meta_cursor_renderer_native_get_instance_private (native);
 
   return priv->gbm;
-}
-
-void
-meta_cursor_renderer_native_get_cursor_size (MetaCursorRendererNative *native,
-                                             uint64_t *width, uint64_t *height)
-{
-  MetaCursorRendererNativePrivate *priv = meta_cursor_renderer_native_get_instance_private (native);
-
-  *width = priv->cursor_width;
-  *height = priv->cursor_height;
 }
 
 void
